### PR TITLE
Adds duck-typing flag; sets value to 'on' whenever it gets set to undefined / null.

### PR DESCRIPTION
**Pull request by bicknellr**
Part of a fix for PolymerElements/paper-toggle-button#66.
Depends on PolymerElements/iron-form#82.

### CODE REVIEW

File: iron-checked-element-behavior.html [hunk 1]
```diff
@@ -83,16 +88,15 @@
       }
     },
 
-    /**
```
Comment (notwaldorf) [inline iron-checked-element-behavior.html:86]: Comment is still good (or if it isn't, you should fix it, not delete it 😄)! We try to comment protected behaviour methods...

File: test/basic.html [hunk 2]
```diff
@@ -99,6 +100,32 @@
...
+      });
+
+      test('value returns "on" when value is set to undefined', function() {
+        var c = fixture('basic');
+
+        c.value = 'abc';
+        assert.equal(c.value, 'abc', 'returns "abc"');
+
+        c.value = undefined;
+        assert.equal(c.value, 'on', 'returns "on"');
```
Comment (notwaldorf) [inline test/basic.html:127]: :gem:

File: iron-checked-element-behavior.html [hunk 3]
```diff
@@ -64,6 +65,10 @@
       '_requiredChanged(required)'
     ],
 
+    created: function() {
+      this._hasIronCheckedElementBehavior = true;
```
Comment (notwaldorf) [inline iron-checked-element-behavior.html:72]: I would maybe add a comment about why this is needed. It doesn't seem to be used anywhere in this class, so I would totally be tempted to nuke it in the future.

Also, I think you might want to define this in the properties block, or else `polylint` will complain that you're accessing a non-defined property (in iron-form, where you're actually using it)

**Comment by notwaldorf** [comment]
Awesome! LGTM, with some nits

File: iron-checked-element-behavior.html (hunk 3 above)
Comment (bicknellr) [inline iron-checked-element-behavior.html:72]: `polylint` checks out, I added the comment though.

File: iron-checked-element-behavior.html (hunk 1 above)
Comment (bicknellr) [inline iron-checked-element-behavior.html:86]: Fixed; also commented `_valueChanged`.

**Pull request merged**